workflow-controller/controller-service/src/main/java/Converters/Converter.java

```diff
@@ -0,0 +1,116 @@
+import com.github.fge.jsonschema.core.report.ProcessingReport;
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.exception.CommonErrorCode;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.models.Activity;
+import io.hyscale.controller.activity.ControllerActivity;
+import io.hyscale.schema.validator.SchemaProcessor;
+import io.hyscale.servicespec.commons.activity.ServiceSpecActivity;
+import io.hyscale.servicespec.commons.builder.ServiceInputType;
+import io.hyscale.servicespec.commons.exception.ServiceSpecErrorCodes;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Converter class implements CommandLine framework ITypeConverter ,
+ * performs command line input validation and assigns to the respective field.
+ */
+public abstract class Converter implements CommandLine.ITypeConverter<List<File>> {
+
+    private static final Logger logger = LoggerFactory.getLogger(Converter.class);
+
+    public abstract String getFilePattern();
+
+    public abstract String getReferenceSchema();
+
+    public abstract String getWarnMessage();
+
+    public abstract int isDataValid(File inputFile) throws Exception;
+
+    public abstract Activity getDataMistmatchActivity();
+
+    /**
+     *  performs command line input validation and assigns to the respective field.
+     *
+     *  1.if input found empty throws Empty file path Hyscale Exception.
+     *  2.if input file does not exits throws File not found Hyscale Exception.
+     *  3.if input is not file throws Cannot process file Hyscale Exception.
+     *  4.if input does not satisfy the respective reference schema throws Invalid Format Hyscale Exception.
+     * @param inputFilePath
+     * @return assigns the file to field if valid.
+     * @throws Exception contains wrapped HyscaleExcpetion for Handling.
+     */
+    @Override
+    public List<File> convert(String inputFilePath) throws Exception {
+
+        List<File> inputFiles = new ArrayList<File>();
+
+        if (StringUtils.isBlank(inputFilePath)) {
+            WorkflowLogger.error(ControllerActivity.EMPTY_FILE_PATH);
+            throw new HyscaleException(CommonErrorCode.EMPTY_FILE_PATH);
+        }
+
+        File file = new File(inputFilePath);
+        if (!file.exists()) {
+            WorkflowLogger.error(ControllerActivity.CANNOT_FIND_FILE, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.FILE_NOT_FOUND);
+        }
+        if (file.isFile()) {
+            String fileName = file.getName();
+            if (!fileName.matches(getFilePattern())) {
+                logger.warn(getWarnMessage(),inputFilePath.substring(inputFilePath.indexOf(ToolConstants.FILE_SEPARATOR) + 1));
```

Comment:
Should we fail in case file name doesnot follow the pattern?

Reply:
If we are not relying on the name of the file anywhere in code, then it can just be a warning , because it is just a recommendation. If we do rely then it is proper to throw an exception.